src/a5/a5_joystick.c

```diff
@@ -98,6 +135,11 @@ static void * a5_joystick_thread_proc(ALLEGRO_THREAD * thread, void * data)
                     }
                     break;
                 }
+                case ALLEGRO_EVENT_JOYSTICK_CONFIGURATION:
+                {
+                    a5_reconfigure_joysticks();
```

Comment:
maybe should call `al_reconfigure_joysticks` just before? I'm not sure.

Reply:
Actually ... I think when to reconfigure is supposed to be controlled by the application, so probably the right thing to do on this event is nothing.

Instead, should this reconfiguration code should be exposed in `a5alleg.h` ?

Reply:
If we were going to implement handling of joystick hot plugging, we would probably want to do it by setting up an event forwarder. Under Allegro 5, it is up to the user to re-evaluate the joystick info after a hot plugging event. There's not really a way to implicitly handle this within Allegro Legacy. Allegro 4 doesn't support this anyway, so the user should not expect this functionality when using Allegro Legacy.

A good way to do this would probably be to add `all_set_event_forward_queue(ALLEGRO_EVENT_QUEUE *)`. When a forwarding queue is set, all internal events generated from Allegro 5 that are used by Allegro Legacy could be forwarded to the user through their supplied queue. This would give the user a way to react to Allegro 5 events such as `ALLEGRO_EVENT_JOYSTICK_CONFIGURATION`.

I think this type of functionality would greatly extend the capabilities of Allegro Legacy without adding a lot of extra work. I'll make an issue of it for now.